tests/test_density_matrix.py

```diff
@@ -455,6 +628,44 @@ def qft_circ():
         np.testing.assert_allclose(dm_backend.state.fidelity(noisy_state), dm_backend.state.fidelity(exact_qft_state))
 
 
+def rand_herm(l: int):
+    """
+    generate random hermitian matrix of size l*l
+    """
+    tmp = np.random.rand(l, l) + 1j * np.random.rand(l, l)
+    return tmp + tmp.conj().T
+
+
+def rand_unit(l: int):
+    """
+    generate random unitary matrix of size l*l from hermitian matrix
+    """
+    return scipy.linalg.expm(1j * rand_herm(l))
+
+
+class TestUtilities(unittest.TestCase):
+
+    # not 2**n as for QM but doesn't matter.
+    def test_rand_herm(self):
+        tmp = rand_herm(np.random.randint(2, 20))
+        np.testing.assert_allclose(tmp, tmp.conj().T)
+
+    def test_rand_unit(self):
+        d = np.random.randint(2, 20)
+        tmp = rand_unit(d)
+
+        # check by applying to a random state
+        # can compare both vectors directly since no global phase introduced in the computation.
+        psi = np.random.rand(d) + 1j * np.random.rand(d)
+        psi /= np.sqrt(np.sum(np.abs(psi) ** 2))
+        np.testing.assert_allclose(tmp @ tmp.conj().T @ psi, psi)
+        np.testing.assert_allclose(tmp.conj().T @ tmp @ psi, psi)
+
+        # direct assert equal identity doesn't seem to work. Precision issues?
+        # np.testing.assert_allclose(tmp @ tmp.conj().T, np.eye(d))
+        # np.testing.assert_allclose(tmp.conj().T @ tmp, np.eye(d))
+
+
 if __name__ == "__main__":
     np.random.seed(2)
```

Comment:
perhaps better to remove seeding?

Reply:
I think it was there before I jumped in. Maybe it's better for reproducibility?